pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1480,22 +1485,77 @@ public void openLedgerFailed(ManagedLedgerException exception, Object ctx) {
+                                                    ZkIsolatedBookieEnsemblePlacementPolicy.class);
+                                    if (localPolicies.isPresent()
+                                            && localPolicies.get().bookieAffinityGroup != null) {
+                                        Map<String, Object> properties = Maps.newHashMap();
+                                        properties.put(ZkIsolatedBookieEnsemblePlacementPolicy
+                                                        .ISOLATION_BOOKIE_GROUPS,
+                                                localPolicies.get().bookieAffinityGroup
+                                                        .getBookkeeperAffinityGroupPrimary());
+                                        properties.put(ZkIsolatedBookieEnsemblePlacementPolicy
+                                                        .SECONDARY_ISOLATION_BOOKIE_GROUPS,
+                                                localPolicies.get().bookieAffinityGroup
+                                                        .getBookkeeperAffinityGroupSecondary());
+                                        managedLedgerConfig.
+                                                setBookKeeperEnsemblePlacementPolicyProperties(properties);
+                                    } else if (namespace.getTenant().equals("pulsar")) {
+                                        Map<String, Object> properties = Maps.newHashMap();
+                                        properties.put(ZkIsolatedBookieEnsemblePlacementPolicy
+                                                .ISOLATION_BOOKIE_GROUPS, "*");
+                                        properties.put(ZkIsolatedBookieEnsemblePlacementPolicy
+                                                .SECONDARY_ISOLATION_BOOKIE_GROUPS, "*");
+                                        managedLedgerConfig.
+                                                setBookKeeperEnsemblePlacementPolicyProperties(properties);
+                                    } else {
+                                        Map<String, Object> properties = Maps.newHashMap();
+                                        properties.put(ZkIsolatedBookieEnsemblePlacementPolicy
+                                                .ISOLATION_BOOKIE_GROUPS, "");
+                                        properties.put(ZkIsolatedBookieEnsemblePlacementPolicy
+                                                .SECONDARY_ISOLATION_BOOKIE_GROUPS, "");
+                                        managedLedgerConfig.
+                                                setBookKeeperEnsemblePlacementPolicyProperties(properties);
+                                    }
+                                }
+                            }
+                        } catch (Exception e) {
+                            log.error("Error getting bookie isolation info from zk");
+                            throw new IllegalStateException(e);
```

Comment:
this is  from  eolivelli's review.
we should fail,
otherwise the broker will continue with a wrong configuration
and we will push the data to the wrong bookies probably

Reply:
I think this would not make the broker shut down